trl/trainer/grpo_trainer.py

```diff
@@ -868,8 +868,24 @@ def _generate_and_score_completions(
             self._total_train_tokens += self.accelerator.gather_for_metrics(attention_mask.sum()).sum().item()
         self._metrics[mode]["num_tokens"] = [self._total_train_tokens]
 
-        completion_length = self.accelerator.gather_for_metrics(completion_mask.sum(1)).float().mean().item()
-        self._metrics[mode]["completion_length"].append(completion_length)
+        # log completion lengths, mean, min, max
+        arg_completion_mask = self.accelerator.gather_for_metrics(completion_mask.sum(1))
+        mean_completion_length = arg_completion_mask.float().mean().item()
+        min_completion_length = arg_completion_mask.float().min().item()
+        max_completion_length = arg_completion_mask.float().max().item()
+        self._metrics[mode]["mean_completion_length"].append(mean_completion_length)
+        self._metrics[mode]["min_completion_length"].append(min_completion_length)
+        self._metrics[mode]["max_completion_length"].append(max_completion_length)
+
+        # identify sequences that terminated with EOS and log their lengths
+        agg_terminated_with_eos = self.accelerator.gather_for_metrics(is_eos.any(dim=1))
```

Comment:
Since we now have the sequences with / without EOS, WDYT about also logging `clip_ratio_completion_length` or `truncated_completion_length_ratio` as suggested in [SimpleRL-Zero](https://arxiv.org/pdf/2503.18892):

![Screenshot 2025-03-31 at 11 24 36](https://github.com/user-attachments/assets/8e86a627-e182-4a9a-821e-b8b2a5007c13)

Reply:
This is just  `num_terminated_completions / num_total_completions` ?

Reply:
No it's `num_truncated_completions / num_total_completions` (i.e. a high clip ratio means a lot of completions are too verbose)